src/FwdState.cc

```diff
@@ -748,20 +745,21 @@ FwdState::connectDone(const Comm::ConnectionPointer &conn, Comm::Flag status, in
         if (originWantsEncryptedTraffic && // the "encrypted traffic" part
                 !peer->options.originserver && // the "through a proxy" part
                 !peer->secure.encryptTransport) // the "exclude HTTPS proxies" part
-            return establishTunnelThruProxy();
+            return establishTunnelThruProxy(conn);
     }
 
-    secureConnectionToPeerIfNeeded();
+    secureConnectionToPeerIfNeeded(conn);
 }
 
 void
-FwdState::establishTunnelThruProxy()
+FwdState::establishTunnelThruProxy(const Comm::ConnectionPointer &conn)
 {
     AsyncCall::Pointer callback = asyncCall(17,4,
                                             "FwdState::tunnelEstablishmentDone",
                                             Http::Tunneler::CbDialer<FwdState>(&FwdState::tunnelEstablishmentDone, this));
     HttpRequest::Pointer requestPointer = request;
-    const auto tunneler = new Http::Tunneler(serverConnection(), requestPointer, callback, connectingTimeout(serverConnection()), al);
+    const auto tunneler = new Http::Tunneler(conn, requestPointer, callback, connectingTimeout(serverConnection()), al);
+    tunneler->usesPconn_ = true;
```

Comment:
This hack is ugly on many levels: Tunneler does not really know or need to know anything about persistent connections. Sooner or later, some caller is likely to forget to set usedPconn_. It is not clear when `usesPconn_` should be made true; do the rules differ from one caller to another? Etc. However, I do not think we should fix this problem in _this_ PR. Let's attack it separately in a different, dedicated PR that will come after this one.

```suggestion
    tunneler->usesPconn_ = true; // TODO: Replace this hack with proper Connection-Pool association
```

Reply:
commit 91061de3bd

The Http::Tunneler::usesPconn_ hack is used also for HappyConnOpener (allowPconn_ flag) and used inside this patch for Ssl::PeerConnector class to. 
Currently only the FwdState class uses these flags to update its fwdPconnPool statistics of type PconnPool. 
The other callers does not use persistent connections (like the tunnel.cc) or they are not use PconnPool objects but implement their own persistent connections pools like the ICAP subsystem does.